### ISSUE TITLE
Test failure and recovery cases for block with withheld data

### ISSUE THREAD

**Issue by musalbas** [celestiaorg/test-infra#21]
We should do network tests for cases like if a block producer withholds some of the block:
(1) will full nodes reconstruct it in case the block is recoverable, and how long will it take?
(2) after the block is reconstructed - what happens to light clients that have hanging DAS queries?
(3) if the erasure code is valid, will a bad coding fraud proof be generated and how long?

**Comment by liamsi** [comment]
> (1) will full nodes reconstruct it in case the block is recoverable, and how long will it take?

I think it is worth to consider worst-case scenarios here were essentially there are only (and theoretically sufficiently many) light nodes that all sampled successfully and a full node that is trying to recover the full block an is only connected to a subset of these light nodes (but not other full nodes that already have and serve the data).

**Comment by musalbas** [comment]
Yes, that's a good idea. Formulas for computing the minimum light nodes required are in the fraud proofs paper, and I also have code for it.
